hoomd/md/methods/methods.py

```diff
@@ -30,7 +32,104 @@ def _detach_hook(self):
         self._simulation.state.update_group_dof()
 
 
-class NVT(Method):
+class Thermostat(_HOOMDBaseObject):
+    def __init__(self, kT):
+        param_dict = ParameterDict(kT=Variant)
+        param_dict["kT"] = kT
+        self._param_dict.update(param_dict)
+        self._cpp_obj = None
+
+    @hoomd.logging.log(requires_run=True)
+    def thermostat_energy(self):
+        """Energy the thermostat contributes to the Hamiltonian \
+        :math:`[\\mathrm{energy}]`."""
+        return self._cpp_obj.getThermostatEnergy(self._simulation.timestep)
+
+    def _attach_hook(self):
+        pass
+
+
+class ConstantEnergy(Thermostat):
+    def __init__(self):
+        super().__init__(1.0)
+
+    def _attach_hook(self):
+        self._cpp_obj = _md.Thermostat(self.kT)
+
+class MTTKThermostat(Thermostat):
+    def __init__(self, kT, tau):
+        super(MTTKThermostat, self).__init__(kT)
+        param_dict = ParameterDict(tau=float(tau),
+                                   translational_thermostat_dof=(float, float),
+                                   rotational_thermostat_dof=(float, float))
+        param_dict.update(
+            dict(translational_thermostat_dof=(0, 0),
+                 rotational_thermostat_dof=(0, 0))
+        )
+        self._param_dict.update(param_dict)
+
+    def _attach_hook(self):
+        self._cpp_obj = _md.MTTKThermostat(self.kT, self.tau)
+
+    def thermalize_thermostat_dof(self):
+        r"""Set the thermostat momenta to random values.
+
+        `thermalize_thermostat_dof` sets a random value for the momentum
+        :math:`\xi`. When `Integrator.integrate_rotational_dof` is `True`, it
+        also sets a random value for the rotational thermostat momentum
+        :math:`\xi_{\mathrm{rot}}`. Call `thermalize_thermostat_dof` to set a
+        new random state for the thermostat.
+
+        .. important::
+            You must call `Simulation.run` before `thermalize_thermostat_dof`.
+            Call ``run(steps=0)`` to prepare a newly created `hoomd.Simulation`.
+
+        .. seealso:: `State.thermalize_particle_momenta`
+        """
+        if not self.is_attached:
+            raise StandardError("Call run(0) before attempting to thermalize the MTTK thermostat")
+        self._simulation._warn_if_seed_unset()
+        self._cpp_obj.thermalizeThermostat(self._simulation.timestep)
+
+
+class BussiThermostat(Thermostat):
+    def __init__(self, kT):
+        super(BussiThermostat, self).__init__(kT)
+
+    def _attach_hook(self):
+        self._cpp_obj = _md.BussiThermostat(self.kT)
+
+
+class BerendsenThermostat(Thermostat):
+    def __init__(self, kT, tau):
+        super(BerendsenThermostat, self).__init__(kT)
+        param_dict = ParameterDict(tau=float(tau))
```

Comment:
```suggestion
        param_dict = ParameterDict(tau=float)
```
You set `tau` to the given value on the next line.